x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/relevance/curations/Condition.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.relevance.curations;
+
+import org.elasticsearch.xpack.relevancesearch.query.RelevanceMatchQueryBuilder;
+
+/**
+ * Conditions used to match a curation. This is a base class that can be implemented for different types of conditions.
+ */
+public abstract class Condition {
+
+    private final String value;
+
+    protected Condition(String value) {
+        if (value == null) {
+            throw new IllegalArgumentException("Value must be specified");
+        }
+        this.value = value;
+    }
+
+    public static Condition buildCondition(String context, String value) {
+        if (context == null) {
+            throw new IllegalArgumentException("Context must be specified");
+        }
+        return switch (context) {
+            case "query" -> new QueryCondition(value);
+            default -> throw new IllegalArgumentException("Unknown context: " + context);
+        };
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public abstract boolean match(RelevanceMatchQueryBuilder queryBuilder);
+
+    /**
+     * A condition that relies on the query exactly matching the value
+     */
+    public static class QueryCondition extends Condition {
+
+        public QueryCondition(String value) {
+            super(normalizeValue(value));
```

Comment:
I'm torn on calling `normalizeValue` in the constructor. While we currently don't have any use for the raw value, we could consider either: 
- Storing a normalized copy of the value in an adjacent member
- Normalizing the value when we call `match` 

On the other hand I realize this is being rather nitpicky and it's also probably fine as-is.

Reply:
Either way is fine; this is just an example on what we could do, and not a real normalization (trim extra spaces, do lowercase with locale...) 

I think it's enough for the PoC, and we might want to go deeper for an actual implementation